app/templates/components/github-repo.hbs

```diff
@@ -9,8 +9,16 @@
         {{#if showSettings}}
           <a data-test-close-repo-settings>Collapse</a>
         {{else}}
-          {{#if projectGithubRepo}}
-            <a data-test-open-repo-settings-edit>Edit settings</a>
+          {{#if hasProject}}
+            {{#if belongsToProject}}
+              <a data-test-open-repo-settings-edit>Edit settings</a>
+            {{else}}
+              <div data-test-repo-project class="github-repo__project">
+                <span>Connected to</span>
+                <span><img class="icon" src={{githubRepo.project.iconThumbUrl}} /></span>
+                <span>{{githubRepo.project.title}}</span>
+              </div>
+            {{/if}}
```

Comment:
This is all totally untested right now, as is the changed visual state of the repo.